kiwi/system/uri.py

```diff
@@ -91,38 +92,41 @@ def translate(self):
             raise KiwiUriStyleUnknown(
                 'URI scheme not detected {uri}'.format(uri=self.uri)
             )
-
-        if uri.scheme == 'obs' and self.repo_type == 'yast2':
-            return self._obs_distribution(
-                ''.join([uri.netloc, uri.path])
-            )
         elif uri.scheme == 'obs':
-            return self._obs_project(
-                ''.join([uri.netloc, uri.path])
-            )
+            if check_build_environment and Defaults.is_buildservice_worker():
+                return self._buildservice_path(
+                    name=''.join([uri.netloc, uri.path]),
+                    fragment=uri.fragment,
+                    urischeme=uri.scheme
+                )
+            else:
+                if self.repo_type == 'yast2':
```

Comment:
As discussed in IRC I doubt we should consider this case... specially inside the buildservice, inside the buildservice I would just use the provided <download_url> from the buildservice, any case that can't be handled with that parameter I would just not consider it.